frontend/src/components/AttendanceDashboard.js

```
@@ -208,13 +209,53 @@ const AttendanceDashboard = ({ userRole }) => {
         bulkStatus, 
         {} // Empty attributes for bulk update since we're only setting status
       );
       
+      // If changing to holiday status, process holiday payment adjustments
+      if (bulkStatus === 'holiday') {
+        try {
+          console.log(`Processing holiday payment adjustments for ${selectedDate.toLocaleDateString()}`);
+          
+          // Add a small delay and process only once per date
+          const dateKey = selectedDate.toDateString();
+          if (window.holidayProcessingInProgress && window.holidayProcessingInProgress[dateKey]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)